### PR TITLE
Included Python3.12 support

### PR DESCRIPTION
**Pull request by RobertSamoilescu**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@ jobs:
     strategy:
       matrix:
         os: [ ubuntu-latest ]
-        python-version: ['3.9', '3.10', '3.11']
+        python-version: ['3.9', '3.10', '3.11', '3.12']
         include: # Run windows tests on only one python version
           - os: windows-latest
             python-version: '3.11'
```
Comment (sakoush) [inline .github/workflows/ci.yml:41]: should this be 3.12 as well?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -35,7 +35,7 @@ jobs:
     strategy:
       matrix:
         os: [ ubuntu-latest ]
-        python-version: ['3.9', '3.10', '3.11']
+        python-version: ['3.9', '3.10', '3.11', '3.12']
```
Comment (sakoush) [inline .github/workflows/ci.yml:38]: do we also support 3.13? if so we can add it here for the tests?

**Review by sakoush — APPROVED** [review]
LGTM

File: .github/workflows/ci.yml (hunk 2 above)
Comment (RobertSamoilescu) [reply .github/workflows/ci.yml:38]: I would rather do it in a separate PR.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (RobertSamoilescu) [reply .github/workflows/ci.yml:41]: Maybe another PR. Not really a priority atm

**Pull request merged**